sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventData.java

```diff
@@ -256,6 +233,31 @@ public int hashCode() {
         return Arrays.hashCode(body);
     }
 
+    /**
+     * A specified key-value pair of type {@link Context} to set additional information on the event.
+     *
+     * @return the {@link Context} object set on the event
+     */
+    Context getContext() {
+        return context;
+    }
+
+    /**
+     * Adds a new key value pair to the existing context on Event Data.
+     *
+     * @param key The key for this context object
+     * @param value The value for this context object.
+     * @throws NullPointerException if {@code key} or {@code value} is null.
+     * @return The updated {@link EventData}.
+     */
+    EventData addContext(String key, Object value) {
```

Comment:
`getContext()` should be sufficient as the users can just do `getContext().addData(key, value).addData(key, value)`. Do we need this `addContext()` method in `EventData`?

Reply:
The unfortunate thing is that addContext returns a new object, so they wouldn't be able to do the assignment back to `this.context`